File: app/src/main/java/app/fit/fitndflow/ui/features/common/notification/NotificationPermissionDialog.java
package app.fit.fitndflow.ui.features.common.notification;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.viewbinding.ViewBinding;

import com.fit.fitndflow.R;
import com.fit.fitndflow.databinding.DialogNotificationPermissionBinding;

import app.fit.fitndflow.ui.features.common.CommonDialogFragment;

public class NotificationPermissionDialog extends CommonDialogFragment implements View.OnClickListener {
    private DialogNotificationPermissionBinding binding;
    public static String TAG = "NotificationDialog";


    @Override
    protected int getBorderType() {
        return NAKED;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = super.onCreateView(inflater, container, savedInstanceState);
        initListeners();
        return view;
    }

    @Override
    public ViewBinding getBinding() {
        binding = DialogNotificationPermissionBinding.inflate(getLayoutInflater());
        return binding;
    }


    private void initListeners(){
        binding.buttonPanel.setOnClickListener(this);
        binding.closeButton.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {

            switch (view.getId()) {
                case R.id.buttonPanel:
                    MyNotificationManager.requestPermission(requireActivity());
                    break;
            }
            dismissAllowingStateLoss();
        }
}

File: app/src/main/java/app/fit/fitndflow/ui/features/common/notification/NotificationPermissionDialog.kt
package app.fit.fitndflow.ui.features.common.notification

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import app.fit.fitndflow.ui.features.common.BorderType
import app.fit.fitndflow.ui.features.common.CommonDialogFragment
import com.fit.fitndflow.databinding.DialogNotificationPermissionBinding

class NotificationPermissionDialog : CommonDialogFragment() {
    companion object {
        @JvmField
        val TAG = "NotificationDialog"
    }
    private lateinit var binding: DialogNotificationPermissionBinding

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = super.onCreateView(inflater, container, savedInstanceState)
        initListeners()
        return view
    }
    private fun initListeners(){
        binding.apply {
            buttonPanel.setOnClickListener {
                MyNotificationManager.requestPermission(requireActivity())
                dismissAllowingStateLoss()
            }
            closeButton.setOnClickListener { dismissAllowingStateLoss() }
        }
    }

    override fun getBorderType(): BorderType {
        return BorderType.NAKED
    }

    override fun getBinding(): ViewBinding {
        binding = DialogNotificationPermissionBinding.inflate(layoutInflater)
        return binding
    }
}